### ISSUE TITLE
Add snmp-inventory of UPS

### ISSUE THREAD

**Issue by vollkommenIrrelevant** [glpi-project/glpi-agent#247]
### Documentation

Yes, I read it

### Is your feature request related to a problem?

[https://forum.glpi-project.org/viewtopic.php?id=285107](https://forum.glpi-project.org/viewtopic.php?id=285107) => Post 3

netdiscovery & netinventory tasks will really only report networking devices and printers. This is because we are really limited from the snmp point of view. This means GLPI will see your UPS as a networking device (and indeed it is some kind of as it answers to SNMP).
So the rule which will apply will be the network device ones and you should see them in "network devices" asset panel.
UPS is even not a supported asset type in GLPI.

### Expected behavior

UPS are not detected as Networkdevice. They are imported as UPS directly into glpi.

### Alternatives you've considered

-

**Comment by vollkommenIrrelevant** [comment]
Ps. Our Riello-UPS are holding Information/Alert when to change the battery. It would be good If this Information can be stored somehwere in glpi.

**Comment by fabrepe** [comment]
Hello there,

I have the same issue with Eaton ePDU equipments which are imported as network devices.
The mapping between SNMP equipment and device type seems to be statically defined here :
* https://github.com/glpi-project/glpi-agent/blob/05278aa37fafa5b43cc2ae5710a78753f9f71c1a/lib/GLPI/Agent/Tools/Hardware.pm#L51
* https://github.com/glpi-project/glpi-agent/blob/7e6018f0c1461c1150ad2aa95d4dab9fb252b796/share/sysobject.ids#L3822

Maybe just modifying the mapping would solve the problem.

Regards,

**Comment by g-bougard** [comment]
Hi @fabrepe 
as I said on the forum and @vollkommenIrrelevant used my words in the description: this is firstly a GLPI issue as we don't have UPS type support during automatic inventory. So changing agent code or sysobject.ids database won't help until such type is supported server-side.

**Comment by fabrepe** [comment]
My bad, I didnt see the point.

thank you for your quick reply.

Regards

**Comment by Megachip** [comment]
Maybe this will solved with the support of GenericObjects ...

**Comment by g-bougard** [comment]
Yes, and you still can try it with [GLPI 11.0.0-beta2](https://github.com/glpi-project/glpi/releases/tag/11.0.0-beta2).

**Comment by g-bougard** [comment]
Hi @vollkommenIrrelevant ,

GLPI 11 was released few months ago. It supports a rule engine and you should now be able to handle an UPS type GLPI side.

So I'm closing this issue. Feel free to open a new issue if you don't manage to use this GLPI feature to achieve your goal.
